File: src/noise-modulated-sines.hs
-- noise modulated sines (jmcc)

import Sound.OpenSoundControl
import Sound.SC3.ID
import Sound.SC3.Lang.Events.OverlapTexture

nms :: UGen
nms =
    let f = midiCPS (rand 'a' 60 100)
        o = fSinOsc AR (mce2 f (f + 0.2)) 0
        a = lfNoise2 'b' KR (f * mce2 0.15 0.16) * 0.1
    in o * a

cmb :: Synthdef
cmb =
    let i = in' 2 AR 0
    in synthdef "cmb" (replaceOut 0 (combN i 0.3 0.3 4 + mceReverse i))

act :: Transport t => t -> IO ()
act fd = do
  _ <- async fd (d_recv cmb)
  send fd (g_new [(2,AddToTail,0)])
  send fd (s_new "cmb" (-1) AddToTail 2 [])
  play fd (overlapTextureU' (4,4,4,maxBound) nms)

main :: IO ()
main = withSC3 act

File: gr/noise-modulated-sines.fs
\ noise modulated sines (jmcc) #6

: f 60 100 Rand.ir MIDICPS ;
: nsm f f 0.2 + 2 mce 0 FSinOsc.ar f 0.15 0.16 2 mce * LFNoise2.kr 0.1 * * ;
: pp { z } z 0.3 0.3 4 CombN.ar z unmce swap 2 mce + ;
' pp texture-post-proc
nsm 4 4 4 inf overlap-texture
